src/lib.rs

```diff
@@ -125,6 +126,21 @@ pub fn crls(
     })
 }
 
+/// Return an iterator certificate revocation lists (CSRs) from `rd`.
+///
+/// Filters out any PEM sections that are not CSRs and yields errors if a problem occurs
+/// while trying to extract a CSR.
+#[cfg(feature = "std")]
+pub fn csrs(
```

Comment:
Do you actually have a use case for this? I don't think we should just add public API because it exists for some of the other item types. I also think, unlike for certificates, that storing multiple CSRs in a single stream might be uncommon.

Reply:
Not really, I've just tried to fit with the existing APIs. I also think that the most common use case is to have one CSR at a time. Maybe we could provide a function that yields the first CSR it encounters, like the [`rustls_pemfile::private_key`](https://docs.rs/rustls-pemfile/latest/rustls_pemfile/fn.private_key.html) ?

Reply:
Is that something you would want to use (perhaps in rcgen)?

Reply:
Yep, to replace the [implementation of `CertficateSingingRequestParam::from_pem`](https://github.com/rustls/rcgen/blob/9c5d2eb7e3d0db8198710e382a270afced999fc7/rcgen/src/csr.rs#L38)

Reply:
Okay, then let's do that.

Reply:
Great! I've already pushed the required code ;)